Automattic/jetpack
# Plugins: Add license activation links on upgrade pages

This PR adds a hint on upgrade pages for customers to add a preexisting license key if they have one.

For styling specifications, see `TTzDk3dzct551hYWRXPFa3-fi-6506%3A67001`.
Fixes `p1HpG7-l29-p2`.

## Proposed changes:

### Backup

* Create a new Header component for the admin page, with added logic to detect whether or not to show a link to activate an existing license.
* Refactor `isFullyConnected` into a custom hook, to be used in a couple places.
* Move the `NoBackupCapabilities` component into its own file, for readability.
* Convert a couple else-if-returns to simple if-returns, for readability.

<img width="1580" alt="Screenshot 2023-03-13 at 09 48 35" src="https://user-images.githubusercontent.com/670067/225044389-5d6931f2-312d-4b59-b0e5-fd419485524c.png">

### Boost

* Create a new `ActivateLicense.svelte` component for use in `BenefitsInterstitial.svelte`.
* Add a license activation link on the upgrade/benefits page, with specified styling.
* Update the style of the upgrade/benefits page header and footer so that their backgrounds and alignment match other plugins (white background, left-aligned logo, plenty of vertical padding).
* Remove duplicated terms-of-service text in favor of new TermsOfService component.
* Add SCSS color variables for link colors (blue-50 and blue-60), as specified in styling specs.

<img width="1133" alt="Screenshot 2023-03-13 at 14 56 25" src="https://user-images.githubusercontent.com/670067/225044392-22150312-aa57-4da8-956b-ab19525c0ed0.png">

### Search

* Create a new Header component for the pricing page.
* Add a license activation link on the upgrade/pricing page, with specified styling.

<img width="698" alt="Screenshot 2023-03-10 at 15 09 55" src="https://user-images.githubusercontent.com/670067/225044382-e66a5df3-3de5-42b0-9001-19429ffd1837.png">

### Social

* Create a new Header component for the admin page, with added logic to detect whether or not to show a link to activate an existing license.
* Add a license activation link on the upgrade/pricing page, with specified styling.
* Add a new field to the initial app state, `admin_url`, for the current site's wp-admin root URL; and, create a new selector, `getAdminUrl`, to retrieve it from JavaScript.

<img width="824" alt="Screenshot 2023-03-10 at 15 10 16" src="https://user-images.githubusercontent.com/670067/225044383-8badec6b-8ec4-4b61-bb35-26844efe2ad3.png">

### Other information:


## Jetpack product discussion

`p1HpG7-l29-p2`

## Does this pull request change what data or activity we track or use?

No.

## Testing instructions:

*(See reference screenshots above.)*

1. In your testing environment, ensure you're using a site that does not have any active Jetpack subscriptions but is fully connected with a user account.
2. Install the **Jetpack VaultPress Backup** plugin and visit its dashboard. (You should see a page asking you to upgrade.) Verify you see a link in the page header that directs you where to enter an existing license key, and that the link sends you to the license activation page.
3. Install the **Jetpack Boost** plugin. Visit its dashboard, scroll down, and click the invitation to upgrade. Verify you're sent to an upgrade page, similar to what you saw for Backup. The text and link should be identical in appearance and behavior.
4. Install the **Jetpack Search** plugin and visit its dashboard. (You should see a page asking you to upgrade.) Verify you see a link similar to what's outlined for the other plugins, and that the link and text behave the same as the others.
5. Install the **Jetpack Social** plugin and visit its dashboard. (You should see a page asking you to upgrade.) Verify you see a link similar to what's outlined for the other plugins, and that the link and text behave the same as the others.
